### PR TITLE
MARK-297: adding validateOrder file in flows v1

### PR DESCRIPTION
**Pull request by igorms-pro**

### CODE REVIEW

File: components/asset-actions/buttons/AcceptBuyOfferButton.tsx [hunk 1]
```diff
@@ -38,12 +57,18 @@ export function AcceptBuyOfferButton({
     <TransactionDialogButton
       label="Accept"
       open={open}
-      onOpenChange={setOpen}
+      onOpenChange={(isOpen) => {
+        setOpen(isOpen)
+        if (!isOpen) {
+          setIsValidationLoading(false)
```
Comment (raphael-verdier) [inline components/asset-actions/buttons/AcceptBuyOfferButton.tsx]: Maybe also reset the error message to '' no?

File: components/asset-actions/buttons/BuyAssetButton.tsx [hunk 2]
```diff
@@ -84,20 +115,34 @@ export function BuyAssetButton({
...
         </Case>
         <Case value="buy-quantity">
-          <BuyQuantityStep
-            asset={asset}
-            order={order}
-            quantity={quantity}
-            setQuantity={setQuantity}
-            onValid={nextStep}
-          />
+          {isErc1155 && isValidationLoading ? (
```
Comment (raphael-verdier) [inline components/asset-actions/buttons/BuyAssetButton.tsx]: Why is "isErc1155 " here? Shouldn't the validation also happen for ERC721?

File: components/asset-actions/buttons/AcceptBuyOfferButton.tsx [hunk 3]
```diff
@@ -53,7 +78,20 @@ export function AcceptBuyOfferButton({
...
+            <div className="flex items-center justify-center ">
+              <Loader size={22} className="animate-spin" />
+            </div>
+          ) : errorMessages.title ? (
+            <div className="flex flex-col gap-5 ">
+              <h3 className="w-full text-center text-xl font-semibold">
+                {errorMessages.title}
+              </h3>
+              <p className="w-full text-center">{errorMessages.message}</p>
+            </div>
```
Comment (raphael-verdier) [inline components/asset-actions/buttons/AcceptBuyOfferButton.tsx]: Make this a component and share it with BuyAssetButton

File: components/asset-actions/buttons/BuyAssetButton.tsx [hunk 4]
```diff
@@ -118,13 +163,26 @@ export function BuyAssetButton({
...
-            order={order}
-            onValid={closeDialog}
-            quantity={quantity}
-            transactionPrice={transactionPrice}
-          />
+          {isValidationLoading ? (
+            <div className="flex items-center justify-center ">
+              <Loader size={22} className="animate-spin" />
+            </div>
+          ) : errorMessages.title ? (
```
Comment (raphael-verdier) [inline components/asset-actions/buttons/BuyAssetButton.tsx]: I don't think you need twice this ternary logic. I think it should be possible to add it before the Case components so you have this logic only once

File: lib/web3/hooks/useValidateSellListing.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    if (isOpen) {
+      validate()
+    } else {
+      setValidationResult(null)
+    }
+  }, [order, asset, nftSwapSdk, isErc1155, isOpen])
+
+  return validationResult
```
Comment (raphael-verdier) [inline lib/web3/hooks/useValidateSellListing.ts]: You could have a returned package {validationResult, isLoadingValidation}, this way you don't have to redefine an isLoading boolean in each component using this hook.

File: components/asset-actions/buttons/BuyAssetButton.tsx (hunk 2 above)
Comment (igorms-pro) [reply components/asset-actions/buttons/BuyAssetButton.tsx]: well I agree that `isErc1155`is not mandatory here but we can only have step `buy-quantity` is case of buying a 1155 asset. I put it as double check

File: components/asset-actions/buttons/AcceptBuyOfferButton.tsx (hunk 1 above)
Comment (igorms-pro) [reply components/asset-actions/buttons/AcceptBuyOfferButton.tsx]: nope, it will display the steps before closing

**Pull request merged**